BadgeUpClient/ResourceClients/MetricClient.cs

```diff
@@ -36,5 +38,35 @@ public async Task<List<MetricResponse>> GetAll()
 		{
 			return await this.m_httpClient.GetAll<MetricResponse>(ENDPOINT);
 		}
+
+		/// <summary>
+		/// Retrieves a list of metrics for a single subject.
+		/// </summary>
+		/// <param name="subjectId">Unique subject the metrics is associated with.</param>
+		/// <returns></returns>
+		public async Task<List<MetricResponse>> GetAllBySubject(string subjectId)
+		{
+			var result = await this.m_httpClient.GetAll<MetricResponse>(ENDPOINT + "/" + subjectId);
+			return result;
+		}
+
+		/// <summary>
+		/// Creates a single metric.
+		/// </summary>
+		/// <param name="key">States what the metric represents, such as a player action or other occurrence. You may want to create your keys in groups, e.g. "player:jump".</param>
+		/// <param name="subject">Identifies the subject this metric is for. This would commonly be a unique user identifier.</param>
+		/// <param name="value">Value of this metric.</param>
+		/// <returns></returns>
+		public async Task<MetricResponse> Create(Metric metric)
+		{
+			if (metric == null)
+			{
+				throw new ArgumentNullException(nameof(metric));
+			}
+
+			var request = new MetricRequest(metric);
+
+			return await this.m_httpClient.Post<MetricResponse>(request, ENDPOINT + "/" + metric.Subject);
```

Comment:
We need to make sure the subject is properly url encoded

Reply:
Thanks! I'll make a note to move the URL-encoding process into a single place, so that we don't forget to do that in future tasks. PR updated.